src/ddmd/dmangle.d

```diff
@@ -173,21 +180,123 @@ public:
     }
 
     ////////////////////////////////////////////////////////////////////////////
+    // writes a back reference number with base 10 using
+    //  characters starting from '0' for all digits but the
+    //  last digit. It uses characters starting from lastDigitBase.
+    // This allows detecting the end of the number as well as
+    //  having different type of back references determined by the
+    //  last digit.
+    final void writeBackRefPos(size_t pos, char lastDigitBase)
+    {
+        enum base = 10;
+        size_t mul = 1;
+        while (pos >= mul * base)
+            mul *= base;
+        while (mul >= base)
+        {
+            auto dig = cast(ubyte)(pos / mul);
+            buf.writeByte('0' + dig);
+            pos -= dig * mul;
+            mul /= base;
+        }
+        buf.writeByte(lastDigitBase + cast(ubyte)pos);
+    }
+
+    final bool backrefType(Type t)
+    {
+        if (!t.isTypeBasic())
+        {
+            auto p = cast(size_t*)dmd_aaGet(&types, cast(Key)t);
+            if (*p)
+            {
+                buf.writeByte('Q');
+                writeBackRefPos(buf.offset - 1 - *p, 'a');
+                return true;
+            }
+            *p = buf.offset;
+        }
+        return false;
+    }
+
+    final bool backrefSymbol(Dsymbol s)
```

Comment:
I know this function probably seems obvious since you just wrote it, but in 3 years nobody will remember and trying to figure it out from the function body is a fair amount of work. I.e. please add a Ddoc comment saying what it does.